metrics/metrics.go

```diff
@@ -0,0 +1,177 @@
+		if err != nil {
+			return sender, errors.Wrapf(err, "failed to open file: %s", config.Out)
+		}
+		writer = w
+	}
+
+	sender.logger = log.New(writer, config.Prefix, config.Flag)
+
+	return sender, nil
+}
+
+func (s logSender) Send() {
+	metrics.Log(metrics.DefaultRegistry, s.config.Interval, s.logger)
+}
+
+func newMetricsSender(config config.Metrics) (metricsSender, error) {
+	var metricsSender metricsSender
+	var err error
+
+	switch config.Type {
+	case Log:
+		metricsSender, err = newLogSender(config.Log)
+	default:
+		err = errors.New(fmt.Sprintf("unkown metrics sender type: %s", config.Type))
+	}
+
+	return metricsSender, err
+}
+
+type Metrics struct {
+	config config.Metrics
+	sender metricsSender
+
+	GcLast metrics.Gauge
+	GcNext metrics.Gauge
+	GcNum  metrics.Gauge
+
+	GcPausePerSecond metrics.Gauge
+	GoroutineNum     metrics.Gauge
+
+	HeapAlloc   metrics.Gauge
+	HeapIdle    metrics.Gauge
+	HeapInuse   metrics.Gauge
+	HeapObjects metrics.Gauge
+	HeapSys     metrics.Gauge
+
+	MemoryAlloc   metrics.Gauge
+	MemoryFrees   metrics.Gauge
+	MemoryLookups metrics.Gauge
+	MemoryMallocs metrics.Gauge
+	MemorySys     metrics.Gauge
+
+	StackInUse metrics.Gauge
+
+	Connections metrics.Gauge
+}
+
+func NewMetrics(config config.Config) (*Metrics, error) {
+	m := &Metrics{
+		config:           config.Metrics,
+		GcLast:           metrics.NewGauge(),
+		GcNext:           metrics.NewGauge(),
+		GcNum:            metrics.NewGauge(),
+		GcPausePerSecond: metrics.NewGauge(),
+		GoroutineNum:     metrics.NewGauge(),
+		HeapAlloc:        metrics.NewGauge(),
+		HeapIdle:         metrics.NewGauge(),
+		HeapInuse:        metrics.NewGauge(),
+		HeapObjects:      metrics.NewGauge(),
+		HeapSys:          metrics.NewGauge(),
+		MemoryAlloc:      metrics.NewGauge(),
+		MemoryFrees:      metrics.NewGauge(),
+		MemoryLookups:    metrics.NewGauge(),
+		MemoryMallocs:    metrics.NewGauge(),
+		MemorySys:        metrics.NewGauge(),
+		StackInUse:       metrics.NewGauge(),
+		Connections:      metrics.NewGauge(),
+	}
+
+	metrics.Register("GcLast", m.GcLast)
+	metrics.Register("GcNext", m.GcNext)
+	metrics.Register("GcNum", m.GcNum)
+	metrics.Register("GcPausePerSecond", m.GcPausePerSecond)
+	metrics.Register("GoroutineNum", m.GoroutineNum)
+	metrics.Register("HeapAlloc", m.HeapAlloc)
+	metrics.Register("HeapIdle", m.HeapIdle)
+	metrics.Register("HeapInuse", m.HeapInuse)
+	metrics.Register("HeapObjects", m.HeapObjects)
+	metrics.Register("HeapSys", m.HeapSys)
+	metrics.Register("MemoryAlloc", m.MemoryAlloc)
+	metrics.Register("MemoryFrees", m.MemoryFrees)
+	metrics.Register("MemoryLookups", m.MemoryLookups)
+	metrics.Register("MemoryMallocs", m.MemoryMallocs)
+	metrics.Register("MemorySys", m.MemorySys)
+	metrics.Register("StackInUse", m.StackInUse)
+	metrics.Register("Connections", m.Connections)
+
+	sender, err := newMetricsSender(m.config)
+	if err != nil {
+		return m, err
+	}
+	m.sender = sender
+
+	return m, nil
+}
+
+func (m *Metrics) Start() {
```

Comment:
__[golint]__
* [golint] exported method Metrics.Start should have comment or be unexported

<!-- Comment by [SideCI](https://sideci.com) -->